### PR TITLE
Volatile load bugfix

### PR DESCRIPTION
**Pull request by neildryan**
This fixes the issue with the `striped_hello` spmd test timing out (#114) by making the LLVM pass respect the programmer's use of `volatile`, as well as fixing an issue where the test wasn't agnostic to manycore size.

### CODE REVIEW

File: software/bsg_manycore_lib/bsg_tilegroup.h [hunk 1]
```diff
@@ -104,42 +104,62 @@ void extern_store_char(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset,
 }
 
 
-float extern_load_float(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset) {
+float extern_load_float(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset,
```
Comment (tommydcjung) [inline software/bsg_manycore_lib/bsg_tilegroup.h]: just for my understanding, when is extern_load_float used?

Comment (neildryan) [reply software/bsg_manycore_lib/bsg_tilegroup.h]: When loading a float from another tile -- it's effectively the same as extern_load_int, but clang has much stricter typing requirements than GCC

Comment (tommydcjung) [reply software/bsg_manycore_lib/bsg_tilegroup.h]: okay. so are you saying that calling "A[i][id]" gets translated into extern_load_<type>() by the compiler (in layman's term)?

Comment (neildryan) [reply software/bsg_manycore_lib/bsg_tilegroup.h]: If A[i][id] is an rvalue, then yes! If it's an lvalue, A[i][id] translates into extern_store_() by the LLVM pass.

File: software/spmd/striped_hello/main.c [hunk 2]
```diff
@@ -127,10 +128,10 @@ int main()
 
     int bsg_id = bsg_x * bsg_tiles_X + bsg_y;
```
Comment (tommydcjung) [inline software/spmd/striped_hello/main.c]: BTW, shouldn't it be something like int bsg_id = bsg_Y * bsg_tiles_X + bsg_x;

i think bsg_Y, bsg_X are deprecated (replaced by __bsg_Y or something).
I would refer to bsg_set_tiles_x_y.h. I think it also calculate __bsg_id

Comment (neildryan) [reply software/spmd/striped_hello/main.c]: I'll change the test in another commit
